.githooks/pre-push

```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+cd diagnostics-app
+npm run test:ci
```

Comment:
btw, why you don't run `npm ci` as you do it in `.github/workflows/angular.test.yml`?

Can't we just run `pre-push` script in the workflow?

Reply:
> nit: please add `cd ..` so we always will be in the root after running some block of operations.

This hook will always run from the project root automatically, so we don't need to worry about any operations.

> btw, why you don't run `npm ci` as you do it in `.github/workflows/angular.test.yml`?

That `npm ci` script is only available while an action is being executed. It is used to set up npm for testing purposes during action workflow. It is a mandatory script to be run in an npm action and has nothing to do with our app.

> Can't we just run `pre-push` script in the workflow?

We will later add on linting and style guide checks inside the pre-push workflow which are not needed in the action. All we need in the test action is the npm run test:ci command. Hence it was written that way!

Reply:
> This hook will always run from the project root automatically, so we don't need to worry about any operations.

Imagine when you will add Chrome extension you will need some automation for it as well, so you will need to go back to the root directory.

So you code may look like

```
cd diagnostics-app
npm run test:ci
cd ..

cd chrome-extension
npm run test:ci
cd ..
```

The benefit of `cd ..` is that you can simply comment command for any block and script still will work.

> We will later add on linting and style guide checks inside the pre-push workflow which are not needed in the action. All we need in the test action is the npm run test:ci command.

I'd run all automation scripts including format and style check in git action as well, because git hooks are not required to run before submitting commits. Moreover, you can create PR using github web editor which does not run git hooks.


My main point is that you should have the same script for all possible checks in github actions and in git hooks.

Reply:
> The benefit of `cd ..` is that you can simply comment command for any block and script still will work.

Oh alright! Yes this makes sense. I thought you were talking about `cd ../diagnostics-app`. 

> I'd run all automation scripts including format and style check in git action as well, because git hooks are not required to run before submitting commits. Moreover, you can create PR using github web editor which does not run git hooks.
> 
> My main point is that you should have the same script for all possible checks in github actions and in git hooks.

I missed these scenarios, will create a common npm script for this. :+1:

Reply:
@Lamzin updated the scripts!